vcd/resource_vcd_nsxt_alb_service_engine_group.go

```diff
@@ -0,0 +1,249 @@
+			// Engine Group API endpoint does not return entities once they are consumed, and it is impossible to make
+			// data source.
+			"service_engine_group_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "NSX-T ALB Importable Service Engine Group Name",
+			},
+			"max_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "NSX-T ALB Service Engine Group maximum virtual services",
+			},
+			"reserved_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "NSX-T ALB Service Engine Group reserved virtual services",
+			},
+			"deployed_virtual_services": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "NSX-T ALB Service Engine Group deployed virtual services",
+			},
+			"ha_mode": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "NSX-T ALB Service Engine Group HA mode",
+			},
+			"overallocated": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Boolean value that shows if virtual services are overallocated",
+			},
+			"sync_on_refresh": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Boolean value that shows if sync should be performed on every refresh",
+			},
+		},
+	}
+}
+
+func resourceVcdAlbServiceEngineGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	if !vcdClient.Client.IsSysAdmin {
+		return diag.Errorf("this resource is only supported for Providers")
+	}
+
+	// Lookup Importable Service Engine Group
+	albImportableSeGroup, err := vcdClient.GetAlbImportableServiceEngineGroupByName(
+		d.Get("alb_cloud_id").(string), d.Get("service_engine_group_name").(string))
+	if err != nil {
+		return diag.Errorf("unable to find Importable Service Engine Group by Name '%s': %s",
+			d.Get("service_engine_group_name").(string), err)
+	}
+
+	albSeGroupConfig := getNsxtAlbServiceEngineGroupType(d, albImportableSeGroup.NsxtAlbImportableServiceEngineGroups.ID)
+	createdAlbController, err := vcdClient.CreateNsxtAlbServiceEngineGroup(albSeGroupConfig)
+	if err != nil {
+		return diag.Errorf("error creating NSX-T ALB Service Engine Group '%s': %s", albSeGroupConfig.Name, err)
+	}
+
+	d.SetId(createdAlbController.NsxtAlbServiceEngineGroup.ID)
+
+	return resourceVcdAlbServiceEngineGroupRead(ctx, d, meta)
+}
+
+func resourceVcdAlbServiceEngineGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	if !vcdClient.Client.IsSysAdmin {
+		return diag.Errorf("this resource is only supported for Providers")
+	}
+
+	// If the only value for update is for 'sync_on_refresh' flag - there is no need to perform any API calls
```

Comment:
Shouldn't we return a read instead?

Reply:
Very good catch!
I did check UI and saw sync operations going, but this makes it clearer.